src/components/ThemeToggle.tsx

```diff
@@ -0,0 +1,63 @@
+"use client"
+
+import { useEffect, useState } from "react"
+
+export default function ThemeToggle() {
+  const [theme, setTheme] = useState<"light" | "dark">("light")
+
+  useEffect(() => {
+    // read stored preference
+    const stored = typeof window !== "undefined" ? localStorage.getItem("theme") : null
+    if (stored === "dark" || (stored === null && window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches)) {
+      document.documentElement.classList.add("dark")
+      setTheme("dark")
+    } else {
+      document.documentElement.classList.remove("dark")
+      setTheme("light")
+    }
+  }, [])
+
+  const toggle = () => {
+    if (theme === "dark") {
+      document.documentElement.classList.remove("dark")
+      localStorage.setItem("theme", "light")
+      setTheme("light")
+    } else {
+      document.documentElement.classList.add("dark")
+      localStorage.setItem("theme", "dark")
+      setTheme("dark")
+    }
+    // Force update logo visibility to avoid any CSS ordering/system preference delays
+    try {
+      const light = document.querySelectorAll<HTMLElement>('.logo-light')
+      const dark = document.querySelectorAll<HTMLElement>('.logo-dark')
+      const isDarkNow = document.documentElement.classList.contains('dark')
+      light.forEach(el => { el.style.display = isDarkNow ? 'none' : 'inline-block' })
+      dark.forEach(el => { el.style.display = isDarkNow ? 'inline-block' : 'none' })
```

Comment:
This doesn't look too efficient. I'd rather say that you use [contexts](https://legacy.reactjs.org/docs/context.html) as it'd be more react friendly. This would get rid of all that logo issue I mentioned earlier.
Talking aboit efficiency, I don't think it really matters here either. So do whatever you think is the easiest. Take this as a suggestion to solve the other issues I mentioned tho